### PR TITLE
Dac 3526 ga4 qa v1.0.3 analytics package kbv pii risk orange

### PR DESCRIPTION
**Pull request by Joanneyoung01**
### Description
This will redact any text from the form response tracker if passed the information which states that the information is sensitive.

This has been tested on the Demo app and mobile id check

### Tickets

[https://govukverify.atlassian.net/browse/DAC-3526](https://govukverify.atlassian.net/browse/DAC-3526)

### Steps to Reproduce

On a repository, pass in an additional parameter `isDataSensitive: 'true'` into the options object 

- check out branch, run `npm build`
- Copy contents from `/dist/lib/analytics.js` to the node_modules 

Open your chosen repository to test
(If testing on demo app you can check out branch `DAC-3526-hide-sensitive-data`)

- navigate to `node_modules/@govuk-one-login/one-login-analytics/lib/analytics.js` and paste in the content
- in the base.njk add `isDataSensitive: 'true'` as an option
Screenshot 2024-06-07 at 17 00 59

- Open GTM and run the app

Radio buttons
Screenshot 2024-06-07 at 16 16 44

Checkbox
Screenshot 2024-06-07 at 16 17 01

Free text field
Screenshot 2024-06-07 at 16 17 19

Drop down list
Screenshot 2024-06-07 at 16 17 31

Free text field 
Screenshot 2024-06-07 at 16 17 45

Form change response
Screenshot 2024-06-07 at 16 19 42

Mobile ID check
Screenshot 2024-06-07 at 17 21 58

### CODE REVIEW

File: src/analytics/formResponseTracker/formResponseTracker.ts [hunk 1]
```diff
@@ -116,4 +120,14 @@ export class FormResponseTracker extends FormTracker {
...
       : "undefined";
   }
+
+  /**
+   * Redacts field value if data is flagged as sensitive
+   *
+   * @param {any} string- Text field input
+   * @return {string} The text field or undefined
+   */
+  redactPII(parameter: any): string {
```
Comment (di-aholme) [inline src/analytics/formResponseTracker/formResponseTracker.ts:130]: This param should be able to be a string. Only place we're using `redactPII` is with the `validateParameter` function which looks like it returns a string from following the chain of function calls back to `stripPIIFromString`

**Review by di-aholme — CHANGES_REQUESTED** [review]
Looks good, going to run some local testing now. Left one minor comment

File: src/analytics/formResponseTracker/formResponseTracker.ts (hunk 1 above)
Comment (Joanneyoung01) [reply src/analytics/formResponseTracker/formResponseTracker.ts]: Ah yes good spot! I've updated

File: src/analytics/formTracker/formTracker.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export class FormTracker extends BaseTracker {
   DROPDOWN_FIELD_TYPE = "drop-down list";
   RADIO_FIELD_TYPE = "radio buttons";
 
-  constructor() {
+  constructor(isDataSensitive: boolean = false) {
```
Comment (di-aholme) [inline src/analytics/formTracker/formTracker.ts:9]: Is this still required as the value is set in formResponseTracker? If it is, should it be true?

**Review by di-aholme — CHANGES_REQUESTED** [review]

File: src/analytics/formTracker/formTracker.ts (hunk 2 above)
Comment (Joanneyoung01) [reply src/analytics/formTracker/formTracker.ts]: ah this was legacy from when we were thinking of extracting this out of the form response tracker, i've removed!

**Review by nickhealGDS — APPROVED** [review]

**Comment by SophiaDWokoma** [comment]
Tested , with the ipv-cri-kbv-front repo
Screenshot 2024-06-13 at 16 41 18
148252375/800567f2-7beb-46ec-b71b-c018dc9d8cc0">

**Pull request merged**